api/src/main/java/org/pmiops/workbench/google/CloudStorageServiceImpl.java

```diff
@@ -35,6 +39,62 @@ public String readMandrillApiKey() {
     return mandrillKeys.getString("api-key");
   }
 
+  public void copyAllDemoNotebooks(String workspaceBucket)  {
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+    Bucket demoBucket = storage.get(getDemosBucketName());
+    Page<Blob> allBlobs = demoBucket.list();
+    List<Blob> notebookFiles = new ArrayList<>();
+    notebookFiles.addAll(getFileTypeFromPage(allBlobs, ".ipynb"));
+    while (allBlobs.hasNextPage()) {
+      allBlobs = allBlobs.getNextPage();
+      notebookFiles.addAll(getFileTypeFromPage(allBlobs, ".ipynb"));
+    }
+    notebookFiles.forEach((Blob blob) -> {
+      BlobId demoNotebookId = blob.getBlobId();
+      BlobId targetLocation = BlobId.of(workspaceBucket, "notebooks/" + demoNotebookId.getName());
+      copyBlob(demoNotebookId, targetLocation);
+    });
```

Comment:
Previous comment applies this ... It feels like we should be able to `demoBucket.list() -> stream -> filter on name -> new blob -> copy blob.`

Reply:
Was slacking with Bru about this. My first stab at streaming this would look like 
```
    StreamSupport
        .stream(demoBucket.list().getValues().spliterator(), false)
        .filter(blob -> blob.getBlobId().getName().endsWith(".ipynb"))
        .map(blob -> new ImmutableMap.Builder<BlobId, BlobId>()
                .put(blob.getBlobId(), BlobId.of(workspaceBucket, "notebooks/" + blob.getBlobId().getName())).build())
        .forEach(pair -> copyBlob(pair.keySet().asList().get(0), pair.values().asList().get(0)));
```
But I think that can be cleaned up some more as well. There should be something simpler than the `ImmutableMap` to use.